### DISCUSSION TITLE
v3.11.1

### DISCUSSION THREAD

**Discussion by jsjoeio** [coder/code-server#3920] [Announcements]
v3.11.1

VS Code v1.57.1

Upgrading is as easy as installing the new version over the old one. code-server
maintains all user data in `~/.local/share/code-server` so that it is preserved in between
installations.

## Bug Fixes

⭐ Fix server worker path and fix issue with webviews not loading resources! 

- Fix "serviceWorker.js" path #3829 @cuining 
- fix(lib/vscode): patch authority in asWebviewUri #3895 @jsjoeio 

## Documentation

⭐ Updates and a new section for how to uninstall code-server

- chore: update guides.md header to Accessing Web Services #3819 @khorne3
- docs(install): add uninstall instructions #3840 @jsjoeio 

## Development

⭐ Lots of new tests added and a fix to update dependency requirements

- chore: Update dependency requirements. Fix node version manager #3831 @TeffenEllis
- chore: update coverage thresholds #3832 @jsjoeio 
- refactor: only accept string in pathToFsPath #3812 @jsjoeio 
- feat: add registerRequireOnSelf on function #3760 @jsjoeio 
- feat: add tests for shouldEnableProxy #3846 @jsjoeio 
- refactor: clean up test file structure #3866 @jsjoeio 
- feat: add more tests for proxy.ts #3858 @jsjoeio 
- chore: bump tar with resolutions #3899 @jsjoeio 

Cheers! 🍻

<hr /><em>This discussion was created from the release <a href='https://github.com/cdr/code-server/releases/tag/v3.11.1'>v3.11.1</a>.</em>

**Comment by jsjoeio** [comment]
⚠️ Heads-up!

Something went wrong when publishing to npm. We're investigating. See [GH Action](https://github.com/cdr/code-server/runs/3265912373?check_suite_focus=true)

![image](https://user-images.githubusercontent.com/3806031/128570117-a3e35ea4-f616-407a-aa64-61611fb18bf1.png)

**Reply by jsjoeio** [reply]
We were able to reproduce this locally. 

We were able to publish code-server as a [scoped npm package](https://www.npmjs.com/package/@jsjoeio/code-server) by removing `lib/vscode/node_modules.asar`. We will continue investigating.  

This is the symbolic link: `release/lib/vscode/node_modules.asar` -> `node_modules`

**Reply by jsjoeio** [reply]
We don't believe this is an issue with `yarn` since we've seen the error happen in CI with `yarn v1.22.10` and locally with `yarn v1.22.11`

**Reply by jsjoeio** [reply]
We believe that the npm server changed their validation on artifacts to not allow symbolic links. We tested this by downloading code-server 3.10.2 and tried to publish as a scoped package and ran into the same error. 

![image](https://user-images.githubusercontent.com/3806031/128575970-d445e92c-c16d-48da-8ec6-df37994d2049.png)

**Reply by jsjoeio** [reply]
At this point, 3.11.1 is on Docker, but we're still facing issues publishing on npm, Homebrew and AUR. @code-asher is back on Monday and we will tackle this together. Until then, thank you for your patience 😄

**Reply by jsjoeio** [reply]
Note to self: I did install my [scoped published package](https://www.npmjs.com/package/@jsjoeio/code-server) of 3.11.1 locally with `yarn global add @jsjoeio/code-server` which overwrote `code-server` installed with hombrew and it is working as expected.

The only thing different is we removed `lib/vscode/node_modules.asar` from the release before publishing and that allowed us to bypass that npm server error.

![image](https://user-images.githubusercontent.com/3806031/128576591-f319cab0-4ac4-4536-9cfa-611bdf9a7a48.png)

**Reply by jsjoeio** [reply]
Heads-up everyone! Someone from GitHub reached out and informed us that symlinks in npm packages _did indeed_ change 🎉 We're working with them to see what they suggest as a workaround.

I did also check in with @code-asher who said:
>  we `rm` this symlink in the post-install and recreate it (to make sure it's correct) so the packaged link will never be used

We're going to open a PR to remove it and add more discussion there. Thanks again for your patience as we figure this out!

**Reply by jsjoeio** [reply]
https://github.com/cdr/code-server/pull/3935

**Reply by jsjoeio** [reply]
Update this is now available on npm and soon will be available on homebrew: https://github.com/Homebrew/homebrew-core/pull/83111

**Comment by ahmadyahya11** [comment]
![image](https://user-images.githubusercontent.com/34031327/128639143-cc3ac588-3e0a-4aaf-bc8c-55022d3787f1.png)
![image](https://user-images.githubusercontent.com/34031327/128639159-ba0bc15c-062e-4305-8ce9-b5aba3468ad6.png)
Jupyter: Create New Blank Notebook results in the above error.

**Reply by jsjoeio** [reply]
This was an issue before 3.11.1 was released. Please see here: https://github.com/cdr/code-server/issues/3874

We still need to investigate and find a fix

**Comment by rampd2** [comment]
There is still no update to the docker image ([`codercom/code-server`](https://hub.docker.com/r/codercom/code-server)).

Will there be one or what is the current way to use code-server in a docker environment?

Edit: It is but **not additionally tagged as `:latest`**, see:
![image](https://user-images.githubusercontent.com/12480131/132107421-3ec74388-55ca-4d76-92fb-f1643bff61e2.png)

**Reply by jsjoeio** [reply]
Thanks for flagging that @DanielRamp!

@code-asher do you have access to Docker to mark 3.11.1 as latest? Not sure how we missed that

**Reply by code-asher** [reply]
I pushed it manually just now. 

We still have not updated the Docker script to push to latest:

https://github.com/cdr/code-server/blob/b32b4edf3d1167f87cb3f4aaf1acd3d5378cefb3/ci/release-image/docker-bake.hcl#L15
